gno.land/cmd/gnoland/config_set.go

```diff
@@ -0,0 +1,253 @@
+		},
+	)
+
+	return cmd
+}
+
+func execConfigEdit(cfg *configCfg, io commands.IO, args []string) error {
+	// Load the config
+	loadedCfg, err := config.LoadConfigFile(cfg.configPath)
+	if err != nil {
+		return fmt.Errorf("unable to load config, %w", err)
+	}
+
+	// Make sure the edit arguments are valid
+	if len(args) != 2 {
+		return errInvalidConfigSetArgs
+	}
+
+	var (
+		key   = args[0]
+		value = args[1]
+	)
+
+	// Update the config field
+	if err := updateConfigField(
+		loadedCfg,
+		key,
+		value,
+	); err != nil {
+		return fmt.Errorf("unable to update config field, %w", err)
+	}
+
+	// Make sure the config is now valid
+	if err := loadedCfg.ValidateBasic(); err != nil {
+		return fmt.Errorf("unable to validate config, %w", err)
+	}
+
+	// Save the config
+	if err := config.WriteConfigFile(cfg.configPath, loadedCfg); err != nil {
+		return fmt.Errorf("unable to save updated config, %w", err)
+	}
+
+	io.Printfln("Updated configuration saved at %s", cfg.configPath)
+
+	return nil
+}
+
+// updateFieldAtPath updates the field at the given path, with the given value
+func updateConfigField(config *config.Config, key, value string) error {
+	// Get the config value using reflect
+	configValue := reflect.ValueOf(config).Elem()
+
+	// Get the value path, with sections separated out by a period
+	path := strings.Split(key, ".")
+
+	field, err := getFieldAtPath(configValue, path)
+	if err != nil {
+		return err
+	}
+
+	// We've reached the actual field, check if it can be updated
+	if !field.CanSet() {
+		return fmt.Errorf("unable to set value")
+	}
+
+	// Convert the value to the field's type
+	v, err := convertStringToType(value, field.Interface())
+	if err != nil {
+		return fmt.Errorf("unable to convert value to field type, %w", err)
+	}
+
+	// Update the field value
+	field.Set(reflect.ValueOf(v))
+
+	return nil
+}
+
+// generateInvalidFieldError generates an invalid field error
+func generateInvalidFieldError(field string, value reflect.Value) error {
+	var (
+		valueType = value.Type()
+		numFields = value.NumField()
+	)
+
+	fields := make([]string, numFields)
+
+	for i := 0; i < numFields; i++ {
+		fields[i] = valueType.Field(i).Name
+	}
+
+	return fmt.Errorf(
+		"field \"%s\", is not a valid configuration key, available keys: %s",
+		field,
+		fields,
+	)
+}
+
+// convertStringToType attempts to convert the given
+// string value to an output type.
+// Because we opted to using reflect instead of a flag-based approach,
+// arguments (always strings) need to be converted to the field's
+// respective type, if possible
+func convertStringToType(value string, outputType any) (any, error) {
+	parseInt := func(size int) (any, error) {
+		castValue, err := strconv.ParseInt(value, 10, size)
+		if err != nil {
+			return nil, fmt.Errorf("unable to convert to int%d, %w", size, err)
+		}
+
+		return castValue, nil
+	}
+
+	parseUint := func(size int) (any, error) {
+		castValue, err := strconv.ParseInt(value, 10, size)
```

Comment:
Did you mean `strconv.ParseUint`?

Reply:
Yes 🤦‍♂️ 

I've applied @thehowl's suggestions, and dropped this method as part of:
[c09b7ec](https://github.com/gnolang/gno/pull/1605/commits/c09b7ec2a9b0e684c5eee7e65151d530930a3b92)